core/src/main/java/org/mule/runtime/core/internal/processor/simple/ParseTemplateProcessor.java

```diff
@@ -41,8 +56,14 @@ private void loadContentFromLocation() throws InitialisationException {
       if (location == null) {
         throw new IllegalArgumentException("Location cannot be null");
       }
-      content = IOUtils.getResourceAsString(location, this.getClass());
-
+      content = getResourceAsString(location, this.getClass());
+      if (outputMimeType == null) {
+        MediaType fromLocationMediaType = parse(mimetypesFileTypeMap.getContentType(location));
+        //This is because BINARY is the default value returned if nothing can be resolved. We should not force the value.
+        if (!BINARY.equals(fromLocationMediaType)) {
```

Comment:
Is this really so wrong? You’ll be using ANY if not BINARY

Reply:
Isn't that better? I don't know what could happen if BINARY is the mimeType. Maybe that would force some logic elsewhere when in fact we don't know if it should